azkaban-exec-server/src/main/java/azkaban/container/FlowContainer.java

```diff
@@ -617,4 +621,41 @@ public static void deleteSymlinkedFile(final Path symlinkedFilePath)
       throw new ExecutorManagerException(e);
     }
   }
+
+  /**
+   * Handles the FLOW_FINISHED event and initiates the shutdown sequence.
+   * @param event
+   */
+  @Override
+  public void handleEvent(final Event event) {
+    if (event.getType() == EventType.FLOW_FINISHED) {
+      final FlowRunner eventFlowRunner = (FlowRunner)event.getRunner();
+      // Sanity check, assert that the event is for same flow execution
+      if (eventFlowRunner.getExecutionId() != this.flowRunner.getExecutionId()) {
+        logger.warn("Execution id mismatch! Expected execution id: {}, got execution id: {}",
+                eventFlowRunner.getExecutionId(), this.flowRunner.getExecutionId());
+      } else {
+        logger.info("Flow execution finished for execution id: {}",
+                eventFlowRunner.getExecutionId());
+      }
+      shutdown();
```

Comment:
If flow finished event is not for same exec, even then shutting down?
Actually, I understand this situation should not occur on flow container. But just wondering.

Reply:
Does not apply anymore, resolving.